### ISSUE TITLE
NSURLSession return NSURLErrorDomain Code=-1005 "The network connection was lost." 

### ISSUE THREAD

**Issue by kata017** [rs/SDWebImage#1851]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

IOS 10.3 if I would like to download more images with gzip - the NSURLSession return NSURLErrorDomain Code=-1005. Strange but sometimes image downloaded but not show, because these error. 

Can you help me?

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by mythodeia** [comment]
which version of SDWebImage are you using? Can this issue be replicated in the demo project?

**Comment by ligun123** [comment]
I have the same question.
Can you help us?

**Comment by kata017** [comment]
I use 3.7.6 in SDWebImage. And i can replicated in your demo project.

**Comment by mythodeia** [comment]
@kata017 can you upgrade to the newest version and try again?

**Comment by kata017** [comment]
I upgrade the new version. But i received the same error

**Comment by mythodeia** [comment]
you said you can replicate it in the demo project. are you having problems with your internet connection? see [this](https://forums.developer.apple.com/thread/23773)

**Comment by kata017** [comment]
My network connection is good, I am not use proxy.

**Comment by kata017** [comment]
If I change HTTPHeaders - don't receive NSURLErrorDomain Code=-1005 error. But sometimes give a NSURLErrorDomain Code=-1008 "resource unavailable" error. 

 _HTTPHeaders = [@{@"Accept": @"image/*;q=0.9", @"Accept-Encoding": @""} mutableCopy];

**Comment by Feng-xp** [comment]
I have the same question. both 3.7.6 and 4.1.0
receive -1005 error, but imageData size is ok.

**Comment by Feng-xp** [comment]
solved by using keep-alive
